### PR TITLE
CLDR-16022 Add test to detect new aliases for BCP47 keys or non-tz values

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-16022

- [x] This PR completes the ticket.

Add a test that detects any new aliases for BCP47 keys, or for BCP47 types other than tz types (for tz types the alias contains the tzdata name so any new zones will require an alias); these should not be added. This test depends on having the cldr-archive available (since it compares BCP47 aliases in the current version to those in the previous version), and is skipped if the flag is set indicating that the archive is not available.

This also adds the unit test TestBCP47 to TestAll(); there are probably more unit tests that should be added to TestAll(), filed [CLDR-18301](https://unicode-org.atlassian.net/browse/CLDR-18301) about that.

ALLOW_MANY_COMMITS=true

[CLDR-18301]: https://unicode-org.atlassian.net/browse/CLDR-18301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestBCP47.java [hunk 1]
```diff
@@ -422,6 +424,73 @@ public void TestMu() {
             typesFound.add(type);
         }
 
-        assertEquals("mu values are only temperature units for now", allowedBcp47Units, typesFound);
+        // Currently <convertUnits> - and consequently allowedBcp47Units - has a temperature unit
+        // rankine which is not supported for mu, and hence not in typesFound.
+        // Filed CLDR-18300 about this but meanwhile typesFound just needs to be a subset of
+        // allowedBcp47Units.
```
Comment (pedberg-icu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestBCP47.java:430]: see [CLDR-18300](https://unicode-org.atlassian.net/browse/CLDR-18300)

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAll.java [hunk 2]
```diff
@@ -185,6 +185,7 @@ public TestAll() {
                     "org.unicode.cldr.unittest.TestAnnotations",
                     "org.unicode.cldr.unittest.TestAttributeValues",
                     "org.unicode.cldr.unittest.TestBasic",
+                    "org.unicode.cldr.unittest.TestBCP47",
```
Comment (pedberg-icu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAll.java:188]: see  [CLDR-18301](https://unicode-org.atlassian.net/browse/CLDR-18301) "TestAll() in unittest/TestAll.java does not include all of the tests in unittest; should it?"

Comment (macchiati) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAll.java:188]: Also https://unicode-org.atlassian.net/browse/CLDR-17708 — mostly the same?

**Review by macchiati — APPROVED** [review]

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAll.java (hunk 2 above)
Comment (pedberg-icu) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestAll.java:188]: Yes, thanks, I resolved CLDR-18301 as a duplicate of CLDR-17708 and copied the linked issues over.
